fix(Extract from File Node): Detect file encoding

dana-gill opened the pull request:
## Summary

Unless users sets the encoding themselves, we automatically assume that the file being passed in is `utf8` encoding, which is not always true. In this PR, we will use `chardet` to detect file encodings.

Recommended workflow for testing:

```
{
  "nodes": [
    {
      "parameters": {},
      "type": "n8n-nodes-base.manualTrigger",
      "typeVersion": 1,
      "position": [
        0,
        0
      ],
      "id": "64686c0a-64a4-4a33-9e70-038c9d23c25b",
      "name": "When clicking ‘Test workflow’"
    },
    {
      "parameters": {
        "operation": "text",
        "options": {}
      },
      "type": "n8n-nodes-base.extractFromFile",
      "typeVersion": 1,
      "position": [
        460,
        -40
      ],
      "id": "aaac18d3-1e99-4c47-9de8-2dc8bf95abd7",
      "name": "Extract from File"
    },
    {
      "parameters": {
        "fileSelector": "/Users/dana/Dana/bugs/utf_8_bug",
        "options": {}
      },
      "type": "n8n-nodes-base.readWriteFile",
      "typeVersion": 1,
      "position": [
        220,
        -40
      ],
      "id": "5f3d6bcd-1290-4444-85ef-49fea0b3caa7",
      "name": "Read/Write Files from Disk"
    }
  ],
  "connections": {
    "When clicking ‘Test workflow’": {
      "main": [
        [
          {
            "node": "Read/Write Files from Disk",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "Extract from File": {
      "main": [
        []
      ]
    },
    "Read/Write Files from Disk": {
      "main": [
        [
          {
            "node": "Extract from File",
            "type": "main",
            "index": 0
          }
        ]
      ]
    }
  },
  "pinData": {}
}
```

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/NODE-1767/community-issue-extract-from-text-file-corrupts-non-utf8-files
Closes https://github.com/n8n-io/n8n/issues/10950

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
   **Remember, the title automatically goes into the changelog.
   Use `(no-changelog)` otherwise.**
-->
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. <!--
   A bug is not considered fixed, unless a test is added to prevent it from happening again.
   A feature is not complete without tests.
-->
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

Commit:
Add tests to ExtractFromFile

netroy commented on packages/nodes-base/nodes/Files/ExtractFromFile/actions/moveTo.operation.ts:
```diff
@@ -121,8 +122,8 @@ export async function execute(
 
-			const encoding = (options.encoding as string) || 'utf8';
 			const buffer = await this.helpers.getBinaryDataBuffer(itemIndex, binaryPropertyName);
+			const encoding = (options.encoding as string) || (chardet.detect(buffer) as string);
```
considering how many other nodes do something similar, my preference would be to add a `this.helpers.detectBinaryEncoding` in core. 
Also, there might be couple of places in core that could also benefit from having auto-detect for encoding.

Commit:
Add chardet to core

Commit:
Create a helper function in core to detect binary encodings

Commit:
fix(Extract from File Node): Detect file encoding

Unless users sets the encoding themselves, we automatically assume that
the file being passed in is `utf8` encoding, which is not always true.

netroy reviewed (dismissed):
🎉

netroy committed:
Update pnpm-lock.yaml

netroy committed:
Merge remote-tracking branch 'origin/master' into node-1767-extract-from-text-file-corrupts

netroy committed:
avoid using ranges for dependency versions

janober commented:
Got released with [`n8n@1.73.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.73.0)
